src/pudl_archiver/archivers/validate.py

```diff
@@ -285,3 +286,58 @@ def _validate_xml(buffer: BytesIO) -> bool:
         return False
 
     return True
+
+
+def validate_data_continuity(new_datapackage: DataPackage) -> DatasetSpecificValidation:
+    """Check that the archived data are continuous and complete."""
+    description = "Test that data are continuous and complete"
+    success = True
+    note = []
+    part_range_dict = {"year_quarter": [1, 4, 7, 10], "year_month": list(range(1, 13))}
+    # Make a dictionary of part type and parts (ex: {"quarter_year": [1995q1, 1995q2]}) from the new
+    # archive datapackage to make it easier to loop through.
+    parts_dict_list = [resource.parts for resource in new_datapackage.resources]
+    # Identify the newest year for the whole archive
+    newest_year_part = max(
```

Comment:
This assumes that all data sources have partitions with a date in them, which isn't always the case. I think we need to either config each dataset to specify what partition is getting checked, or set a list of partitions that are configured to work with this test (e.g. `year_month`, `year_quarter`) and only test those.